CONTRIBUTING.md

```diff
@@ -0,0 +1,26 @@
+# Contributing
+
+Ember component patterns aims to provide a baseline documentation of best practices for building interactive user interfaces using Ember JS. By highlighting these practices, this repo also is intended to highlight and discourage the use of anti-patterns in Ember apps.
+
+Contributions of all types are welcome. Whether filing issues, testing components against screen readers, submitting examples of anti-patterns or contributing to the code base, all PRs are welcome. It is preferred that each PR be linked to a documented issue, though this is not strictly neccessary if there is no issue for a given PR.
+
+## Getting started
+
+In order to contribute to this repo, please follow the following steps.
+1. Install [`ember-cli`](https://cli.emberjs.com/release/) globally.
+1. Clone this repo.
+1. `cd ember-component-patterns` and `npm install` dependencies.
+1. Run `ember s` from within the repo to serve the project locally.
+
+## Guidelines for contribution
+### JS based PRs
+- Contributions to this repo must follow the [Ember Octane](https://emberjs.com/editions/octane) paradigm.
+- Each new file - whether a route, component, template or controller - should be generated using the `ember g` command.
+- New functionality should be accompanied by a useful test built off the test generated by Ember. Run `npm test` from within the repo to run the test suite locally.
+- JS files must pass the accompanying eslint rules. Run `npm run lint:js` from within the repo to check linting locally.
```

Comment:
It would probably be good to include template linting here too. What do you think?

Reply:
Agreed, I'll make that update and change the command to npm run lint.

Reply:
Actually, I'll add a section for template based contributions.